src/anbox/config.cpp

```diff
@@ -88,3 +84,24 @@ anbox::SystemConfiguration& anbox::SystemConfiguration::instance() {
   static SystemConfiguration config;
   return config;
 }
+
+anbox::SystemConfiguration::SystemConfiguration() {
+  auto gen_resource_path = [] () -> fs::path {
+    const auto snap_path = utils::get_env_value("SNAP");
+    if (!snap_path.empty()) {
+      return fs::path(snap_path) / "usr/share/anbox";
+    }
+
+    const std::string exe = utils::process_get_exe_path(::getpid());
+    const std::size_t pos = exe.rfind("/bin/anbox");
+    if (pos != std::string::npos) {
+      const std::string leading_path = exe.substr(0, pos);
+      return fs::path(leading_path) / "share/anbox";
+    } else {
+      return "/usr/local/share/anbox";
+    }
+  };
+
+  resource_path = gen_resource_path();
+  data_path = "/var/local/lib/anbox";
```

Comment:
See comment above.

Reply:
thank you for the comments, like above, not sure how to have a consistent directory hierarchy for both `snap` and `make install` which requires no further configuration of build or install.